### PR TITLE
Selinux Enable

### PR DESCRIPTION
**Pull request by magicalyak**
This uses a custom policy for selinux, we may need to add some more sanity checks and testing but I default to false with a new variable so it's not enabled by default.
https://github.com/nginxinc/ansible-role-nginx-app-protect/issues/3

### CODE REVIEW

**Comment by magicalyak** [comment]
I've tested this and this works on variations I have.  The default is nginx_selinux: false so users won't use this unless they add that variable

**Comment by magicalyak** [comment]
Going to make one more change - need to respect existing selinux policy unless we set the flag for nginx_selinux.

File: defaults/main.yml [hunk 1]
```diff
@@ -87,3 +87,6 @@ log_policy_syslog_target: 127.0.0.1:514
 
 nginx_demo_workload_protocol: http://
 nginx_demo_workload_host: 10.1.1.1:8080
+
+# Enable enforcing selinux (you may need to open ports on your own)
+nginx_selinux: false
```
[the remark this answers is not in the document]
Comment (magicalyak) [reply defaults/main.yml:92]: It defaults to false. It’s pretty much just centos and rhel and maybe Amazon Linux and suse so I didn’t say centos. If it’s false nap still functions.

File: files/my-appprotect.te [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	class netlink_audit_socket nlmsg_relay;
+	class sock_file write;
+	class security compute_av;
+	class dir create;
+}
+
+#============= audisp_t ==============
+
+#!!!! WARNING: 'unlabeled_t' is a base type.
+#!!!! The file '/etc/ld.so.cache' is mislabeled on your system.
```
[the remark this answers is not in the document]
Comment (magicalyak) [reply files/my-appprotect.te:35]: It’s from bdagent handling that. Ideally the rpm should set the selinux stuff on install. Also ideally bdagent runs as nginx user like nginx.

File: tasks/prerequisites/setup-selinux.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  shell:
+    cmd: cat /var/log/audit/audit.log | audit2allow -M local
+    chdir: /tmp/
+  args:
+    executable: /bin/bash
+
+- name: "(Install: SELinux: Custom) Apply local policy"
+  command: semodule -i /tmp/local.pp
+
+# - name: "(Install: SELinux: Custom) Copy custom policy"
```
[the remark this answers is not in the document]
Comment (magicalyak) [reply tasks/prerequisites/setup-selinux.yml:101]: I will remove them.

File: defaults/main.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (magicalyak) [reply defaults/main.yml:92]: What is the error message can you post it

[the remark this answers is not in the document]
Comment (aknot242) [reply defaults/main.yml:92]: running `systemctl status nginx.service` and it returns the following:

```APP_PROTECT failed to open /opt/app_protect/config/config_set.json```

[the remark this answers is not in the document]
Comment (aknot242) [reply defaults/main.yml:92]: And this appears to be the failure in the playbook run that may be the core issue. This is on CentOS 7.4 in AWS BTW:

```TASK [nginx-app-protect : (Install: SELinux: Custom) Generate policy] **************************************************************************************
fatal: [ec2-35-182-100-54.ca-central-1.compute.amazonaws.com]: FAILED! => {"changed": true, "cmd": "cat /var/log/audit/audit.log | audit2allow -M local", "delta": "0:00:00.368507", "end": "2020-06-06 15:14:52.016562", "msg": "non-zero return code", "rc": 1, "start": "2020-06-06 15:14:51.648055", "stderr": "", "stderr_lines": [], "stdout": "compilation failed:\nlocal.te:6:ERROR 'syntax error' at token '' on line 6:\n\n\n/usr/bin/checkmodule:  error(s) encountered while parsing configuration\n/usr/bin/checkmodule:  loading policy configuration from local.te", "stdout_lines": ["compilation failed:", "local.te:6:ERROR 'syntax error' at token '' on line 6:", "", "", "/usr/bin/checkmodule:  error(s) encountered while parsing configuration", "/usr/bin/checkmodule:  loading policy configuration from local.te"]}```

[the remark this answers is not in the document]
Comment (magicalyak) [reply defaults/main.yml:92]: Ok will take a look in next few days.  It’s a non computing weekend for me so likely Monday ;)

[the remark this answers is not in the document]
Comment (magicalyak) [reply defaults/main.yml:92]: Ok will take a look in next few days.  It’s a non computing weekend for me so likely Monday ;)

[the remark this answers is not in the document]
Comment (magicalyak) [reply defaults/main.yml:92]: I need more information, what is the selinux status on this machine?  We used to set selinux disabled but this is removed, perhaps this is a manual step that must be done (I don't think we should set disabled in our role, that is overreaching IMHO).

**Pull request merged**